src/test/java/com/example/SS2_Backend/service/MTMStableMatchingSolverTest.java

```diff
@@ -0,0 +1,66 @@
+package com.example.SS2_Backend.service;
+
+import com.example.SS2_Backend.dto.mapper.StableMatchingProblemMapper;
+import com.example.SS2_Backend.dto.request.NewStableMatchingProblemDTO;
+import com.example.SS2_Backend.ss.smt.evaluator.impl.TwoSetFitnessEvaluator;
+import com.example.SS2_Backend.util.MatchingProblemType;
+import com.example.SS2_Backend.util.SampleDataGenerator;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.Validator;
+import jakarta.validation.ValidatorFactory;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.moeaframework.core.NondominatedPopulation;
+
+import java.util.Objects;
+import java.util.Set;
+
+import static org.junit.Assert.assertFalse;
+
+public class MTMStableMatchingSolverTest {
+    NewStableMatchingProblemDTO newStableMatchingProblemDTO;
+    SampleDataGenerator sampleData;
+    int numberOfIndividuals1;
+    int numberOfIndividuals2;
+    int numberOfProperties;
+    private Validator validator;
+
+    @BeforeEach
+    public void setUp() {
+        numberOfIndividuals1 = 20;
+        numberOfIndividuals2 = 200;
+        sampleData = new SampleDataGenerator(
+                MatchingProblemType.MTM,
+                numberOfIndividuals1, numberOfIndividuals2,
+                numberOfProperties
+        );
+        newStableMatchingProblemDTO = sampleData.generateDto();
+        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+        validator = factory.getValidator();
+
+    }
+
+    @Test
+    public void testEvaluateFunctions() {
+        newStableMatchingProblemDTO.setEvaluateFunctions(new String[]{"invalid eval func", "invalid eval func"});
+        Set<ConstraintViolation<NewStableMatchingProblemDTO>> violations = validator.validate(newStableMatchingProblemDTO);
+        assertFalse(violations.isEmpty());
+    }
+
+    @Test
+    public void testFitnessCalculation() {
+        TwoSetFitnessEvaluator newEvaluator = new TwoSetFitnessEvaluator(sampleData.generateProblem().getMatchingData());
+        newEvaluator.withFitnessFunctionEvaluation(new double[]{}, sampleData.getFnf());
+    }
+
+    @Test
+    public void testStableMatchingSolver() {
```

Comment:
Đảm bảo case này ko fail nhé, Thêm 2 case cho OTM và OTO tương tự

Reply:
Mà của mình hình như không có service nào là xử lý bài toán OTO cho StableMatching hay sao ấy anh ơi, em đọc trong HomeController thì:
//    @Async("taskExecutor")
//    @PostMapping("/stable-matching-oto-solver")
//    public CompletableFuture<ResponseEntity<Response>> solveStableMatchingOTO(@RequestBody StableMatchingProblemDTO object) {
//        return CompletableFuture.completedFuture(stableMatchingSolver.solveStableMatchingOTO(object));
//    }

Reply:
Khi chạy solver thì cần kiểm tra cái gì để biết là nó qua bài test đấy ạ